pkg/antctl/runtime/runtime.go

```diff
@@ -47,10 +48,24 @@ func ResolveKubeconfig(path string) (*rest.Config, error) {
 			path = clientcmd.RecommendedHomeFile
 		}
 	}
-	if _, err = os.Stat(path); path == clientcmd.RecommendedHomeFile && os.IsNotExist(err) {
-		return rest.InClusterConfig()
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if path == clientcmd.RecommendedHomeFile {
+			config, inClusterErr := rest.InClusterConfig()
+			if inClusterErr == nil {
+				return config, nil
+			}
+			return nil, fmt.Errorf(
+				"failed to resolve kubeconfig: neither a valid kubeconfig file was found at '%s', nor could InClusterConfig be used: %w",
+				path, inClusterErr,
+			)
+		}
+		return nil, fmt.Errorf("failed to resolve kubeconfig: kubeconfig file does not exist at path '%s'", path)
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", path)
```

Comment:
maybe we should use `clientcmd.NewNonInteractiveDeferredLoadingClientConfig` directly (like we do in other parts of the code). it is a bit weird to use `BuildConfigFromFlags`, which itself has a fallback to `inClusterConfig` (even though it should not be used in our case).

Reply:
Updated, thanks!